FluidNC/src/Spindles/HBridgeSpindle.h

```diff
@@ -0,0 +1,104 @@
+    // This adds support for PWM H-Bridge Spindles
+    class HBridgeSpindle : public Spindle {
+    public:
+        HBridgeSpindle() = default;
+
+        // PWM(Pin&& output, Pin&& enable, Pin&& direction, uint32_t minRpm, uint32_t maxRpm) :
+        //     _min_rpm(minRpm), _max_rpm(maxRpm), _output_pin(std::move(output)), _enable_pin(std::move(enable)),
+        //     _direction_pin(std::move(direction)) {}
+
+        HBridgeSpindle(const HBridgeSpindle&) = delete;
+        HBridgeSpindle(HBridgeSpindle&&)      = delete;
+        HBridgeSpindle& operator=(const HBridgeSpindle&) = delete;
+        HBridgeSpindle& operator=(HBridgeSpindle&&) = delete;
+
+        void init() override;
+        void setSpeedfromISR(uint32_t dev_speed) override;
+        void setState(SpindleState state, SpindleSpeed speed) override;
+        void config_message() override;
+        // Configuration handlers:
+        void validate() const override { Spindle::validate(); }
+
+        void group(Configuration::HandlerBase& handler) override {
+            // The APB clock frequency is 80MHz and the maximum divisor
+            // is 2^10.  The maximum precision is 2^20. 80MHz/2^(20+10)
+            // is 0.075 Hz, or one cycle in 13.4 seconds.  We cannot
+            // represent that in an integer so we set the minimum
+            // frequency to 1 Hz.  Frequencies of 76 Hz or less use
+            // the full 20 bit resolution, 77 to 152 Hz uses 19 bits,
+            // 153 to 305 uses 18 bits, ...
+            // At the other end, the minimum useful precision is 2^2
+            // or 4 levels of control, so the max is 80MHz/2^2 = 20MHz.
+            // Those might not be practical for many CNC applications,
+            // but the ESP32 hardware can handle them, so we let the
+            // user choose.
+            handler.item("pwm_hz", _pwm_freq, 1, 20000000);
+            handler.item("output_cw_pin", _output_cw_pin);
+            handler.item("output_ccw_pin", _output_ccw_pin);
+            handler.item("enable_pin", _enable_pin);
+            handler.item("disable_with_s0", _disable_with_zero_speed);
+
+            Spindle::group(handler);
+        }
+
+        // Name of the configurable. Must match the name registered in the cpp file.
+        const char* name() const override { return "HBridgeSpindle"; }
+
+        virtual ~HBridgeSpindle() {}
+
+    protected:
+        // TODO: A/B rename
+        int32_t      _current_pwm_duty;
+        SpindleState _current_state    = SpindleState::Unknown;
+        int          _pwm_cw_chan_num  = -1;
+        int          _pwm_ccw_chan_num = -1;
+        uint32_t     _pwm_period;     // how many counts in 1 period
+        uint8_t      _pwm_precision;  // auto calculated
+        bool         _duty_update_needed = false;
+
+        // _disable_with_zero_speed forces a disable when speed is 0
+        bool _disable_with_zero_speed = false;
+
+        // The parent class does not support a reverse pin
```

Comment:
Eliminate this line and the next because they are confusing.  "Reverse pin" is irrelevant for this hardware.  It has reversing capability, but it has no reverse pin so talking about one is confusing.